crypto/http/http_client.c

```diff
@@ -928,6 +928,23 @@ int OSSL_HTTP_REQ_CTX_nbio_d2i(OSSL_HTTP_REQ_CTX *rctx,
 
 #ifndef OPENSSL_NO_SOCK
 
+static const char *extract_port(const char *host, const char *port, int use_ssl)
+{
+    if (port == NULL) {
+        if (host[0] == '[') { /* IPv6 address enclosed in '[' and ']' */
+            host = strchr(host + 1, ']');
+            if (host == NULL) {
+                ERR_raise(ERR_LIB_HTTP, ERR_R_PASSED_INVALID_ARGUMENT);
+                return NULL;
+            }
+            host++;
+        }
+        if (strchr(host, ':') == NULL)
+            port = use_ssl ? OSSL_HTTPS_PORT : OSSL_HTTP_PORT;
+    }
+    return port;
+}
+
```

Comment:
Please don't reinvent the wheel.

https://github.com/openssl/openssl/blob/master/doc/man3/BIO_parse_hostserv.pod

Reply:
ping @DDvO

Reply:
Oh, I was not aware of `BIO_parse_hostserv()`.
Feels a bit cumbersome to use in this special situation where we just need to find out if any port/service name is included in the string, but anyway using it now.